src/async_std/boxstream.rs

```diff
@@ -285,11 +285,10 @@ where
             self.cipher_len = written;
             self.cipher_off = 0;
 
-            // force the calling async operation to be called again
-            //   so, eventually calls 1
-
-            cx.waker().clone().wake();
-            return Poll::Pending;
+            // Loop over to actually flush the data down to self.stream
+            // FIXME: Rust doesn't optimize tail recursion so this might cause a stack overflow in
+            // extreme circumstances. This should be changed to use a loop.
+            return self.internal_flush(cx);
```

Comment:
Maybe this recursion will never be problematic. `poll_write` should only consume at most `MSG_BODY_MAX_LEN` bytes, so the size of the plaintext internal buffer is bounded, and thus the number of recursive calls to `internal_flush` should also be bounded